web/snapshots_controller_test.go

```diff
@@ -0,0 +1,139 @@
+package web_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/chainlink/store"
+	"github.com/smartcontractkit/chainlink/store/models"
+	"github.com/smartcontractkit/chainlink/utils"
+	"testing"
+
+	"github.com/smartcontractkit/chainlink/internal/cltest"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSnapshotsController_CreateSnapshot_V1_Format(t *testing.T) {
+	t.Parallel()
+	app, cleanup := cltest.NewApplication()
+	defer cleanup()
+
+	j := cltest.FixtureCreateJobWithAssignmentViaWeb(t, app, "../internal/fixtures/web/v1_format_job.json")
+
+	url := app.Server.URL + "/v1/assignments/" + j.ID + "/snapshots"
+	resp := cltest.BasicAuthPost(url, "application/json", bytes.NewBuffer([]byte{}))
+
+	runID := cltest.ParseCommonJSON(resp.Body).ID
+
+	assert.NotNil(t, runID)
+}
+
+func TestSnapshotsController_CreateSnapshot_V1_NotFound(t *testing.T) {
+	t.Parallel()
+	app, cleanup := cltest.NewApplication()
+	defer cleanup()
+
+	url := app.Server.URL + "/v1/assignments/" + "badid" + "/snapshots"
+	resp := cltest.BasicAuthPost(url, "application/json", bytes.NewBuffer([]byte{}))
+	assert.Equal(t, 404, resp.StatusCode, "Response should be not found")
+}
+
+func TestSnapshotsController_CreateSnapshot_V1_LateJob(t *testing.T) {
+	t.Parallel()
+	app, cleanup := cltest.NewApplication()
+	defer cleanup()
+
+	j := cltest.FixtureCreateJobWithAssignmentViaWeb(t, app, "../internal/fixtures/web/v1_format_job_past_endat_time.json")
+
+	url := app.Server.URL + "/v1/assignments/" + j.ID + "/snapshots"
+	resp := cltest.BasicAuthPost(url, "application/json", bytes.NewBuffer([]byte{}))
+
+	assert.Equal(t, 500, resp.StatusCode, "Response should be server error")
+}
+
+func TestSnapshotsController_ShowSnapshot_V1_Format(t *testing.T) {
+	t.Parallel()
```

Comment:
this test was crazy

Reply:
Haha, yeah. It looks like it's testing a lot of the block height confirmation logic, which is probably out of scope for a unit test of the controller. If you assert that the run reaches the PendingConfirmation state, that'd be a sufficient test, and you could cut out a lot of the ethereum mocking.